File: concurrency/application.exs
IO.puts("Starting...")

start_time = :os.system_time(:millisecond)
1..100_000
|> Enum.map(fn _ -> Task.async(fn -> :timer.sleep(100) end) end)
|> Task.await_many(:infinity)

end_time = :os.system_time(:millisecond)

IO.puts("Ended in #{(end_time - start_time) / 1000}s")

File: concurrency/Application.java
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public class Application {
  public static void main(String[] args) {
    long startTime = System.currentTimeMillis();
    System.out.println("Starting");

    List<CompletableFuture<Void>> futures = new ArrayList<CompletableFuture<Void>>();
    for (int i = 0; i < 100000; i++) {
      futures.add(CompletableFuture.runAsync(() -> {
        try {
          Thread.sleep(100);
        } catch (InterruptedException e) {
          e.printStackTrace();
        }
      }));
    }

    CompletableFuture.allOf(futures.toArray(new CompletableFuture[100000])).join();
    
    long endTime = System.currentTimeMillis();
    System.out.println(String.format("Completed in %d s", (endTime - startTime) / 1000));
  }
}